packages/next-international/src/app/client/create-i18n-provider-client.tsx

```diff
@@ -12,13 +12,20 @@ type I18nProviderWrapperProps = {
   children: ReactNode;
 };
 
+export const localesCache = new Map<string, Record<string, unknown>>();
+
 export function createI18nProviderClient<Locale extends BaseLocale>(
   I18nClientContext: Context<LocaleContext<Locale> | null>,
   locales: ImportedLocales,
   fallbackLocale?: Record<string, unknown>,
 ) {
   function I18nProvider({ locale, children }: I18nProviderProps) {
-    const { default: clientLocale } = use(locales[locale as keyof typeof locales]());
+    let clientLocale: any = localesCache.get(locale);
+
+    if (!clientLocale) {
+      clientLocale = use(locales[locale as keyof typeof locales]()).default;
+      localesCache.set(locale, clientLocale);
```

Comment:
I think we can use a `Proxy` to always flatten the locales when `set` is called.

Something like:
```ts
function flattenLocale(obj) {
  // * ...
  return flattenedObj;
}

const cache = new Map();

export const cacheProxy = new Proxy(cache, {
  set: function(target, key, value) {
    const flattenedLocale = flattenLocale(value);
    target.set(key, flattenedLocale);
    return true;
  }
});

cacheProxy.set('en', { nested: { prop: 'value' } });
cacheProxy.get('en');
```

Reply:
Related to https://github.com/QuiiBz/next-international/pull/234#discussion_r1356155015 - using a proxy is overkill and won't be needed later.

Reply:
Yup, sorry. I wasn't aware of the additional complexity (in terms of intermediate calculations) that a Proxy would introduce. I looked at benchmarks afterwards when I made my suggestion, and felt a bit ashamed.

Reply:
That's fine, no worries! Thanks for your help and suggestions.